pkg/build/strategies/layered/layered.go

```diff
@@ -63,14 +64,24 @@ func (b *Layered) CreateDockerfile(config *api.Config) error {
 	}
 
 	buffer.WriteString(fmt.Sprintf("FROM %s\n", b.config.BuilderImage))
-	buffer.WriteString(fmt.Sprintf("COPY scripts %s\n", locations[0]))
+	// if -s / --scripts / --scripts-url are not specified, there will not be a scripts subdir
+	// off of the working dir, so don't copy here, and we'll have to tweak the scripts URL to use the env var
+	// STI_SCRIPTS_PATH
+	scriptsIncluded := len(config.ScriptsURL) > 0
```

Comment:
Yep, revisited the doc, and the three potential sources are listed [here](https://github.com/openshift/source-to-image/blob/master/docs/builder_image.md#s2i-scripts)

For reference, in order of precedence:
1. A script found at the `--scripts-url` URL
2. A script found in the application source `.sti/bin` directory
3. A script found at the default image URL (`io.openshift.s2i.scripts-url` label)

For 3), the value from the label is stored in the config.ScriptURL variable that is checked in layered.go.  Look [here](https://github.com/openshift/source-to-image/blob/master/pkg/build/config.go#L35).  So I think we are fine there, no?

For 2), I *think* if I read [this module](https://github.com/openshift/source-to-image/blob/master/pkg/scripts/install.go#L88) correctly, since the code is part of the source, it will get copied and referenced as part of the COPY src %s line.

I'll try constructing a .sti/bin example on Wednesday and either confirm that interpretation, or address if it goes elsewhere.